### PR TITLE
Items for Overcoming Extreme Climatic Obstacles

### PR DESCRIPTION
**Pull request by ktksan**
This PR adds items for Overcoming Extreme Climatic Obstacles namely : Waterskin, WoolVest, WoodenShoes, ArabicThawb.
The PR only includes the item prefabs and their respective crafting recipes.
Note: The png images added in textures are temporary for testing the droppingand crafting the items. These images need proper artwork which then needs to be added in the ManualLabor.png grid and ManualLabor.atlas and then the prefabs of the items need to be modified accordingly.

### CODE REVIEW

**Comment by ktksan** [comment]
Depends on : https://github.com/Terasology/WildAnimals/pull/56

**Comment by jdrueckert** [comment]
to be honest, I'm not quite sure that the items should reside in `ManualLabor` 🤔 did you take a look at `Equipment` and `EquipmentSmithing`?

**Comment by ktksan** [comment]
The Waterskin item should remain here in my opinion, but the other items could use the equipment module for the EquipmentEffectsSystem etc. 
The crafting recipes should also remain here imo because equipment, smithing modules involve metallic armors/weapons/etc which require smithing. But the items introduced here simply involve knitting/stitching etc for woolVest and arabic Thawb, also manualLabor has strings, cloth,etc which are similar items and woodenSlippers would involve something very similar to what is done in woodencup/bucket. However effects of the items could utilize the equipments module (EquipmentEffectsSystem , Equip item component etc) but.....
It's just that when I enable the equipments module, for some reason the item use speed / attacks speed/ damage or something for all items becomes very very high and simply by keep holding the left click everything in the player's path can be destroyed... So for now I added the items in here.

**Comment by jdrueckert** [comment]
Why should the waterskin stay in `ManualLabor`?
When looking at the module description, it says
> Work is hard. Here are tools a workman needs to make digging and chopping easier.
What does a waterskin have to do with manual labor and the tools to simplify digging and chopping?

Concerning smithing you are totally right, this is certainly not used to create a vest, waterskin or thawb. But tbh I think that's a problem with `EquipmentSmithing` being limited to smithing instead of including other crafts like weaving and leatherworking. So maybe we should rename `EquipmentSmithing` to `EquipmentCrafting` to make it open for other crafts used to make equipment.

How does the item stats seemingly being overriden affect you in working on the equipment for overcoming the climatic obstacles? Sure, if Equipment has that effect, this needs some work and balancing, but it shouldn't block you from making progress with your project.

**Comment by ktksan** [comment]
There reason I think Waterskin (and actually all these recipes) should stay in ManualLabor is not because of the description of the module but because of the current distribution of items being based on the location of the crafting process and items required for them. 
Equipment smithing has a BasicForgingStation for the crafting tool and all the items in there are the ones which require forging. ManualLabor has multiple such crafting ways for eg. AssemblyTable and CrudeAssembly (does not require any such crafting tool). Now the items introduced in this PR are made using the tools in manualLabor as manualLabor already has some items similar to them. 
There's another reason I want waterskin to stay here especially because all the other items would eventually involve having some effects to the player. However, waterskin is a normal essential tool for survival in general(to quench thirst).  The equipment module has a separate whole feature(for the lack of a better word). There's modification of physical stats and magical items and equip slots which are supposed to be only for items which can be equipped(worn) and waterskin is not such an item. 
Also all other items would probably eventually involve adding an effect in AlterationEffects and Equipment but waterskin won't, it's a simple fluid Container.
And please ignore the physical stats comment, that was written before the discussion on discord.

**Comment by jdrueckert** [comment]
> There's another reason I want waterskin to stay here especially because all the other items would eventually involve having some effects to the player. However, waterskin is a normal essential tool for survival in general(to quench thirst). The equipment module has a separate whole feature(for the lack of a better word). There's modification of physical stats and magical items and equip slots which are supposed to be only for items which can be equipped(worn) and waterskin is not such an item.

That indeed is true. Although I'm still convinced that `Equipment` should be of larger scope and also include items that don't have a special effect on the player, how about the following compromise:
Waterskin can stay in `ManualLabor`, all else goes into `Equipment`.

@ktksan Can you live with that? ;)

Just in case, I'll also ping @skaldarnar for information and third opinion.

**Comment by ktksan** [comment]
There are 2 options I have in mind...
1. The recipes stay in ManualLabor and Equipment will have a delta prefab for the items (if that's possible) which adds the effects.. I propose this because as of now all item recipes are present where there respective crafting tools are.. And these items should use the assembly table and crude assembly mode present in manual labor, equipment smithing has only a forging table which is only for metallic tools..
2. We add a dependency (Equipment smithing depends on ManualLabor) and have the items and recipes present in Equipment Smithing but still use the processes and tools defined in ManualLabor.
Which one do you think would be more appropriate?

File: assets/prefabs/items/ArabicThawb.prefab [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "parent": "engine:iconItem",
+    "Item": {
+        "icon": "ManualLabor:ArabicThawb"
+    },
+    "DisplayName": {
+        "name": "Arabic Thawb",
+        "description": "A vest made from cloth which prevents direct exposure to sun and related rise in body temperature."
```
Comment (jdrueckert) [inline assets/prefabs/items/ArabicThawb.prefab]: Same issue as with the items in `WildAnimals`: The item itself has no knowledge about what effect it has... actually there can be situations (`AlterationEffects` not enabled) where the item will have no effect at all...

File: assets/prefabs/items/WoodenShoe.prefab [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "parent": "engine:iconItem",
+    "Item": {
+        "icon": "ManualLabor:WoodenShoe"
+    },
+    "DisplayName": {
+        "name": "Wooden Flat Shoes",
+        "description": "The flat sole on these wooden shoes helps the player walk in the desert without movement speed reduction."
```
Comment (jdrueckert) [inline assets/prefabs/items/WoodenShoe.prefab]: see above

File: assets/prefabs/items/WoolVest.prefab [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+    "parent": "engine:iconItem",
+    "Item": {
+        "icon": "ManualLabor:ManualLabor#WoolVest"
+    },
+    "DisplayName": {
+        "name": "Wool Vest",
+        "description": "A vest made from wool which helps player to prevent hypothermia in cold biomes."
```
Comment (jdrueckert) [inline assets/prefabs/items/WoolVest.prefab]: see above

File: assets/prefabs/items/assemblyProcesses/AssembleWoodenShoe.prefab [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  "Parent": "CrudeAssemblingBaseProcessPart",
+  "ProcessDefinition": {
+    "processType": "ManualLabor:CrudeAssembling"
+  },
+  "RequirementInput": {
+      "requirements": ["CrudeSaw"]
+    },
+  "InventoryInput": {
+    "itemCounts": {
+      "ManualLabor:Plank": 4
```
Comment (jdrueckert) [inline assets/prefabs/items/assemblyProcesses/AssembleWoodenShoe.prefab]: I'd think you'd maybe need some kind of (leather?) straps to fasten the shoe to the foot :thinking:

**Review by jdrueckert — CHANGES_REQUESTED** [review]

File: assets/prefabs/items/assemblyProcesses/AssembleWoodenShoe.prefab (hunk 4 above)
Comment (ktksan) [reply assets/prefabs/items/assemblyProcesses/AssembleWoodenShoe.prefab]: Strings should work I guess... like just tie them to the foot..

**Review by jdrueckert — APPROVED** [review]

**Pull request merged**